example/declarative.dart

```diff
@@ -1,29 +1,31 @@
 // Copyright 2013 Google. All rights reserved. Use of this source code is
 // governed by a BSD-style license that can be found in the LICENSE file.
 
-library grinder_example_1;
+library grinder.example.declarative;
 
 import 'package:grinder/grinder.dart';
 
-void main([List<String> args]) {
-  task('init', init);
-  task('compile', compile, ['init']);
-  task('deploy', deploy, ['compile']);
-  task('default', null, ['deploy']);
-
+void main(List<String> args) {
+  addAnnotatedTasks();
   startGrinder(args);
 }
 
-void init(GrinderContext context) {
+@Task()
+init(GrinderContext context) {
   context.log("I set things up");
 }
 
-void compile(GrinderContext context) {
+@Task(depends: const ['init'])
+compile(GrinderContext context) {
   context.log("I'm compiling now...");
   //context.fail('woot');
 }
 
-void deploy(GrinderContext context) {
+@Task(depends: const ['compile'])
```

Comment:
Thats not bad, but I personally do prefer having a single annotation, maybe I should update the examples to have a similar formatting though?

```dart
@Task(
  depends: const ['foo', 'bar'],
  description: 'My nice task description.')
deploy(GrinderContext context) {
...
}
```
Once we get http://dartbug.com/4046, then we can get rid of `const ` at least.

re: using doc-comments, absolutely, I want to do the same in unscripted:

https://github.com/seaneagan/unscripted/issues/11

but it's blocked on:

http://dartbug.com/15704